### PR TITLE
[kots] Allow users to upload a `.docker/config.json` file

### PR DESCRIPTION
**Pull request by Pothulapati**
## Description

This PR updates the `kots` UI to add a new config option to upload registry
credentials irrespective of the registry being used, which is then merged
into a single `config.json` file and passed as the registry secret which
is then used across the workspace image builder components.

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/12136

## How to test

Upload a `dockerconfigjson` file through the new option, 
Screenshot from 2022-08-18 19-03-10

and see those credentials in the secret by running
```
kubectl -n gitpod get secret builtin-registry-auth --output="jsonpath={.data.\.dockerconfigjson}" | base64 -d
```

## Release Notes

```release-note
[kots] Allow users to upload a `.docker/config.json` file
```

## Werft options:

- [ ] /werft with-preview

### CODE REVIEW

File: install/kots/manifests/gitpod-installer-job.yaml
Comment (Pothulapati) [inline install/kots/manifests/gitpod-installer-job.yaml:187]: Changed this to save the secret in the `gitpod.yaml`, instead of applying so that we can post process it if the user submits more docker configs through the new option

**Comment by lucasvaltl** [comment]
@Pothulapati what does the UX, specifically the user flow, look like here? When does a user set this and how - do they need to also set something else for this to take Affect?

**Comment by Pothulapati** [comment]
@lucasvaltl This adds a new config option in the `registry` section of the `KOTS` UI (irrespective of in-cluster or external registry) through which a user can upload a file.

Screenshot from 2022-08-18 19-03-10

> do they need to also set something else for this to take Affect?

Not really, because this is automatically set everywhere to be used. They will have to obviously use the container image somewhere to actually use the functionality. 🤔

**Comment by Pothulapati** [comment]
/do-not-merge as we discussed some UI improvements to add.

File: install/kots/manifests/gitpod-installer-job.yaml
Comment (MrSimonEmms) [inline install/kots/manifests/gitpod-installer-job.yaml:82]: The `echo "Gitpod: Generate the base Installer config"` line should go above the `/app/installer init > "${CONFIG_FILE}"` so that the logging is representative of what's going on

File: install/kots/manifests/gitpod-installer-job.yaml
Comment (MrSimonEmms) [reply install/kots/manifests/gitpod-installer-job.yaml:187]: Seems a reasonable approach. This means that we'll have a secret at the start of the YAML file, but that shouldn't be a problem as it's what we're doing anyway

File: install/kots/manifests/kots-config.yaml
Comment (MrSimonEmms) [inline install/kots/manifests/kots-config.yaml:112]:
```suggestion
          title: Configure additional registry credentials for pulling workspace images
```

File: install/kots/manifests/kots-config.yaml
Comment (MrSimonEmms) [inline install/kots/manifests/kots-config.yaml:122]:
```suggestion
          help_text: Docker [config JSON file](https://docs.docker.com/engine/reference/commandline/cli/#sample-configuration-file) with auth credentials used to access private registries, for workspace images.
```

**Review by MrSimonEmms — APPROVED** [review]
/hold

Works beautifully. I've respectfully made a couple of suggestions on the config wording, but approved so you can merge when ready

**Comment by Pothulapati** [comment]
This is how the option looks now!

Screenshot from 2022-08-19 18-47-40


Screenshot from 2022-08-19 18-48-11

File: install/kots/manifests/kots-config.yaml
Comment (lucasvaltl) [inline install/kots/manifests/kots-config.yaml:118]:
```suggestion
          title: Private base image registry credentials
```

File: install/kots/manifests/kots-config.yaml
Comment (Pothulapati) [reply install/kots/manifests/kots-config.yaml:122]: > credentials used to access private registries used for pulling base workspace images.

Happy to update, but `used to <> used for <>` sounds a bit confusing?

File: install/kots/manifests/kots-config.yaml
Comment (lucasvaltl) [inline install/kots/manifests/kots-config.yaml:115]:
```suggestion
          help_text: This is useful when you want to use base workspace images in private registries other than the above configured ones or the [default base workspace images stored on Docker Hub.](https://github.com/gitpod-io/workspace-images) 
```

**Review by lucasvaltl — APPROVED** [review]
Added some suggestions that you can merge right in. Otherwise LGTM, although I think we should hack around KOTS a bit to create a better separation between the push and pull registries here.

**Pull request merged**
